GoogleTestAdapter/Core/Runners/CommandLineGenerator.cs

```diff
@@ -69,39 +69,59 @@ private IEnumerable<Args> GetFinalCommandLines(string baseCommandLine)
                 return commandLines;
             }
 
-            string baseCommandLineWithFilter = baseCommandLine + GoogleTestConstants.FilterOption;
-
             List<string> suitesRunningAllTests = GetSuitesRunningAllTests();
-            baseCommandLineWithFilter += GetFilterForSuitesRunningAllTests(suitesRunningAllTests);
-
-            List<TestCase> testCasesNotRunBySuite = GetTestCasesNotRunBySuite(suitesRunningAllTests);
-            List<TestCase> testCasesRunBySuite = _testCasesToRun.Where(tc => !testCasesNotRunBySuite.Contains(tc)).ToList();
-            if (testCasesNotRunBySuite.Count == 0)
+            int suiteLengthAggregate = 0;
+            int groupIdxCounter = 0;
+            int maxSuiteLength = MaxCommandLength - _lengthOfExecutableString - userParam.Length - 1;
+
+            var suiteGroups = suitesRunningAllTests
+                              .Select(suite =>
+                                { suiteLengthAggregate += suite.Length + 3;
```

Comment:
I had to think about the 3 - probably not a good sign. Maybe use a `const int separatorLength = ".*;".Length`?

Reply:
yep!